### PR TITLE
chore: add examples readme and update readme generator

### PR DESCRIPTION
**Pull request by anitarua**
addresses the last task of #56 and #112

### CODE REVIEW

File: Examples/doc-example-apis/readme.swift [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+func main() async {
+  print("Running Momento Cache example!")
+  let cacheName = "example-cache"
+
+  var creds: CredentialProviderProtocol
+    do {
+        creds = try CredentialProvider.fromEnvironmentVariable(envVariableName: "MOMENTO_API_KEY")
+    } catch {
+        print("Error establishing credential provider: \(error)")
+        return
```
Comment (pgautier404) [inline Examples/doc-example-apis/readme.swift:12]: Errors need to exit with a nonzero exit code so CI can flag a failure.
```suggestion
        exit(1)
```

File: Examples/doc-example-apis/readme.swift [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        cacheName: cacheName,
+        key: "key"
+    )
+    switch getResult {
+    case .hit(let hit):
+        print("Cache hit: \(hit.valueString)")
+    case .miss(_):
+        print("Cache miss")
+    case .error(let err):
+        print("Unable to get item in the cache: \(err)")
```
Comment (pgautier404) [inline Examples/doc-example-apis/readme.swift:31]:
```suggestion
        print("Unable to get item in the cache: \(err)")
        exit(1)
```

File: Examples/doc-example-apis/readme.swift (hunk 1 above)
Comment (anitarua) [reply Examples/doc-example-apis/readme.swift:12]: I don't think this file is getting run by ci, I'll move it to a new folder first and exit(1) there

**Review by pgautier404 — DISMISSED** [review]
LGTM!

**Review by pgautier404 — APPROVED** [review]

**Pull request merged**